### PR TITLE
feat: add `airflow.kubernetesPodTemplate.extraContainers`

### LINKED ISSUE #455: Allow users to provide extra Containers in kubernetesPodTemplate
**What is your feature request?**

We like to mount the dag directory into the pod executor. We have our dags in S3 and use s3fs as a sidecar to mount the volume.

**What alternatives have you considered?**

We checked if there are possibilities in Airflow to use this natively, like storing logs in S3.

### PR DESCRIPTION
**Pull request by jonashartwig**
## What issues does your PR fix?

- resolves #455

## What does your PR do?

- Adds the `airflow.kubernetesPodTemplate.extraContainers` value to specify extra containers in the KubernetesExecutor pod_template.

### For all Pull Requests

- [x] Commits are [signed off](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#sign-your-work)
- [x] Commits have [semantic messages](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#semantic-commit-messages)
- [x] Documentation [updated](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#documentation)
- [x] Passes [ct linting](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#linting)

### For releasing ONLY

- [ ] Chart.yaml [version bumped](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#versioning)
- [ ] CHANGELOG.md updated

### CODE REVIEW

File: charts/airflow/files/pod_template.kubernetes-helm-yaml [hunk 1]
```diff
@@ -74,6 +74,10 @@ spec:
       volumeMounts:
         {{- $volumeMounts | indent 8 }}
       {{- end }}
+
```
Comment (thesuperzapper) [inline charts/airflow/files/pod_template.kubernetes-helm-yaml:77]: nit: can you please remove this newline.

File: charts/airflow/values.yaml [hunk 2]
```diff
@@ -328,6 +328,12 @@ airflow:
     ##
     extraVolumes: []
 
+    ## extra containers for the pod template
+    ## - spec for Volume:
+    ##   https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#container-v1-core
+    ##
```
Comment (thesuperzapper) [inline charts/airflow/values.yaml:334]: This docstring is incorrect, can you please update it to:

```yaml
    ## extra containers for the Pod template
    ## - spec for Container:
    ##   https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#container-v1-core
    ##
```

File: charts/airflow/values.yaml [hunk 3]
```diff
@@ -328,6 +328,12 @@ airflow:
...
+    extraContainers: []
```
Comment (thesuperzapper) [inline charts/airflow/values.yaml:335]: nit: I like to keep the `extraVolumeMounts` and `extraVolumes` at the bottom, can you please move `extraContainers` directly above `extraVolumeMounts`?

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
@jonashartwig thanks for the PR!

I have left a few small comments so we can get this merged!

File: charts/airflow/files/pod_template.kubernetes-helm-yaml (hunk 1 above)
Comment (jonashartwig) [reply charts/airflow/files/pod_template.kubernetes-helm-yaml:77]: removed

File: charts/airflow/values.yaml (hunk 3 above)
Comment (jonashartwig) [reply charts/airflow/values.yaml]: corrected

File: charts/airflow/values.yaml (hunk 2 above)
Comment (jonashartwig) [reply charts/airflow/values.yaml]: corrected

**Comment by jonashartwig** [comment]
Than you for taking the time and giving feedback. I adjusted my PR to accompany your requests.

**Comment by thesuperzapper** [comment]
@jonashartwig thanks for your work, I have added this to the `8.6.0` milestone.

**Comment by jonashartwig** [comment]
Happy to hear :)

**Review by thesuperzapper — APPROVED** [review]
This is ready for the `8.6.0` release, and will be merged before we cut it.

**Review by thesuperzapper — APPROVED** [review]

**Pull request merged**